keps/sig-multicluster/5313-placement-decision-api/README.md

```diff
@@ -0,0 +1,412 @@
+metadata:
+  name: app-placement-decision-1
+  namespace: argocd
+  labels:
+	  multicluster.x-k8s.io/placement: app-placement
+	  multicluster.x-k8s.io/scheduler: multicluster-placement-controller
+decisions:
+- clusterName: cluster1
+- clusterName: cluster2
+```
+
+### Terminology
+
+- **Placement**: A scheduler request that asks "where should this workload run?".
+
+- **Scheduler**: Placement controller that writes `PlacementDecisions` based on `ClusterProfiles` and
+  scheduling/placement requirements/specs.
+
+- **Consumer**: Any controller (GitOps engine, workload orchestrator, progressive rollout controller, AI/ML pipeline)
+  that watches `PlacementDecisions` and acts.
+
+### Slicing
+
+* Following [EndpointSlice](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) design,
+  a single Placement can fan out to N `PlacementDecision` slices,
+  each limited to 100 clusters (`EndpointSlice`'s default).
+* All slices for one `Placement` MUST carry the same
+  `multicluster.x-k8s.io/placement=<placement-name>` label so consumers can List with a label selector.
+* The scheduler that owns the slice MUST label it with
+  `multicluster.x-k8s.io/scheduler=<scheduler-name>` for observability.
+
+### Relationship to other SIG-Multicluster (SIG-MC) APIs
+* **ClusterProfile** The inventory. Each decision must reference a matching name `ClusterProfile`
+* **Work API** The workload. A consumer may read `PlacementDecision` then for each cluster creates `Work`.
+
+### User Stories (Optional)
+
+#### Story 1: GPU-aware AI training
+
+* ML scheduler scores every `ClusterProfile` by available GPUs, cost, etc.  
+* Scheduler writes `PlacementDecision` listing the best clusters.
+* A GitOps/Work-API syncer watches the decision and deploys the training job only to the listed clusters.
+* If cost rises or GPUs becoming unavailable, the scheduler updates `PlacementDecision` with new list,
+  the syncer act accordingly.
+
+#### Story 2: Progressive rollout
+
+* A progressive rollout controller begins with some canary clusters.  
+* It creates `PlacementDecision` containing just those clusters.  
+* Gradually updates `PlacementDecision` based on health and then eventually all clusters.  
+* Consumers watch and deploy only where the decision says.  
+
+#### Story 3: Disaster recovery
+
+* A DR controller monitors `ClusterProfiles` status.
+* Controller updates the corresponding `PlacementDecision`,
+  replacing primary cluster with standby cluster.
+* Syncer deletes workloads from the failed cluster and recreates them in the standby cluster.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+* Scope: Namespace scoped for RBAC parity with Work and ClusterProfile.
+
+* The resource is pure data following `EndpointSlice` convention.
+
+* Max size: 100 ClusterDecision entries per slice keeps object well below etcd limit.
+
+* Validation: A webhook may verify that every clusterName exists
```

Comment:
the placement label value should not be empty?
the placement & scheduler label value should be immutable?

Reply:
Done as suggested. Thanks!